tests/test_dns.py

```diff
@@ -40,13 +40,13 @@
 )
 @pytest.mark.asyncio
 async def test_resolve_hostaddr_async_no_resolve(
-    monkeypatch, conninfo, want, env, fail_resolve
+    monkeypatch, conninfo, want, env, fail_resolve, psycopg_dns
 ):
     if env:
         for k, v in env.items():
             monkeypatch.setenv(k, v)
     params = conninfo_to_dict(conninfo)
-    params = await psycopg._dns.resolve_hostaddr_async(params)
+    params = await psycopg_dns.resolve_hostaddr_async(params)
```

Comment:
Honestly I don't understand why this is reported as error by mypy, adding the import in this function:

```python
    from psycopg._dns import resolve_hostaddr_async
```
which says that the module doesn't have that function. However the dependencies are installed and the function works ok in this environment.

It seems a mypy problem, and it takes quite some work to work around it, no?

Reply:
You're right. In fact, I misread the error message initially and thought that the problem was that `psycopg._dns` was not imported in test files (so, that `psycopg` was missing the `_dns` attribute).

Just spend an hour investigating the mypy issue, but it seems tricky, so I'd rather add some `# type: ignore[attr-defined]` and possibly report an issue.